### DISCUSSION TITLE
Babel config

### DISCUSSION THREAD

**Discussion by petarvujovic98** [near/near-sdk-js#229] [Ideas]
Do we want to expose the babel config to the user? Right now any `JavaScript` smart contract project would have a `babel.config.json` file with our preset of plugins and presets. IMO this adds an additional file that the user has in their project, but we don't want them messing with it.

We could just pass the same configuration programmatically to the `babel` call inside of our build script and thus avoid the need for this config file.

If however we have plans to add options that the user could customize through the config file, then it would be sensible to create our own preset so that there is only one _import_ line of our own in that config file.

Please share your thoughts

**Comment by petarvujovic98** [comment]
FYI this is how it would look if we did it the programmatic way:

`cli/cli.js`
```javascript
// start of the file
babel({
  babelHelpers: "bundled",
  extensions: [".ts", ".js", ".jsx", ".es6", ".es", ".mjs"],
  presets: ["@babel/preset-typescript"],
  plugins: [
    "near-sdk-js/lib/build-tools/include-bytes",
    "near-sdk-js/lib/build-tools/near-bindgen-exporter",
    ["@babel/plugin-proposal-decorators", { version: "legacy" }],
  ],
}),
// the rest of the file
```

**Comment by volovyks** [comment]
@petarvujovic98 it's a technical dept that we have. Yes, let's add this config programmatically!
What will happen if the user will add its own config file? Will it override our config? I would say that this flexibility is "nice to have", not a requirement at this point.
